migrations/m171018_174853_content_share.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+use app\migrations\BaseMigration;
+use app\models\ContentShare;
+
+class m171018_174853_content_share extends BaseMigration
+{
+    public function safeUp()
+    {
+        $this->createTable('{{%content_share}}', [
+            'object_type_id' => $this->smallInteger()->notNull(),
```

Comment:
Status is needed for the case tweeting failed.

Reply:
Ideally, the `status` is needed.

I refused it, because in the job I put all the parameters (`objectTypeId`, `objectId`, `serviceId`). You did through the `primary key`, so without status in any way.

I add an entry to the table, only in case of successful sending. Of course errors are written in the log, but who reads it)

I should have re-created jobs in case of failure, but did not)

Reply:
May be handy: https://github.com/yiisoft/yii2-queue/blob/master/src/RetryableJob.php

Reply:
But instead I'd use `content_share` table for the purpose to get stats of what failed exactly.